packages/apollo/src/commands/service/check.ts

```diff
@@ -602,32 +597,77 @@ export default class ServiceCheck extends ProjectCommand {
         )
       );
     } else if (shouldOutputMarkdown) {
-      const serviceName = config.service && config.service.name;
+      // This _should_ always be here; but TypeScript tells us that's optional. If we check it here, then
+      // passing `config` to any other function will signify that `config.service` might now be null or
+      // undefined. Save it as a const to tell TypeScript `service` can't be changed.
+
+      const { service } = config;
+      if (!service) {
+        throw new Error(
+          "Service mising from config. This should have been validated elsewhere"
+        );
+      }
 
-      if (!serviceName) {
+      const graphName = config.service && config.service.name;
+
+      if (!graphName) {
         throw new Error(
-          "The service name should have been defined in the Apollo config and validated when the config was loaded. Please file an issue if you're seeing this error."
+          "The graph name should have been defined in the Apollo config and validated when the config was loaded. Please file an issue if you're seeing this error."
+        );
+      }
+
+      if (compositionErrors) {
+        if (!serviceName) {
+          throw new Error(
+            "Composition errors should only occur when `serviceName` is present. Please file an issue if you're seeing this error."
+          );
+        }
+
+        return this.log(
+          formatCompositionErrorsMarkdown({
+            compositionErrors,
+            graphName,
+            serviceName,
+            tag: config.tag
+          })
         );
       }
 
       return this.log(
         formatMarkdown({
           checkSchemaResult,
+          graphName,
           serviceName,
           tag: config.tag
         })
       );
     }
 
-    this.log(formatHumanReadable({ checkSchemaResult }));
+    if (compositionErrors) {
+      // Add a cosmetic line break
+      console.log("");
+
+      cli.table(compositionErrors, {
```

Comment:
Just curious, does the heroku table and the cli table display tables differently?

Reply:
Not sure! I havent seen the output of heroku table. This was moved from https://github.com/apollographql/apollo-tooling/pull/1378/files#diff-c61805bdfe85787383d4ae9927bd0774L567 to keep the non json/markdown behavior the same